### PR TITLE
Adding example of headOption operator

### PR DESCRIPTION
**Pull request by muskan3006**
References #25468

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/stream/operators/SinkDocExamples.java [hunk 1]
```diff
@@ -114,6 +114,15 @@ static void foldExample() {
     // #fold
   }
 
+  static void headOptionExample() {
+    // #headoption
+    Source<Integer, NotUsed> source = Source.empty();
+    CompletionStage<Optional<Integer>> result = source.runWith(Sink.headOption(), system);
+    result.thenAccept(System.out::println);
+    // None
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/stream/operators/SinkDocExamples.java:122]: I think toString on empty `Optional` will be `Optional.empty`

File: akka-docs/src/test/scala/docs/stream/operators/sink/HeadOption.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import scala.concurrent.{ ExecutionContextExecutor, Future }
+
+object HeadOption {
+  implicit val system: ActorSystem = ???
+  implicit val ec: ExecutionContextExecutor = system.dispatcher
+  def headOptionExample: Future[Unit] = {
+    //#headoption
+    val source = Source.empty
+    val result: Future[Option[Int]] = source.runWith(Sink.headOption)
+    result.map(println)
```
Comment (johanandren) [inline akka-docs/src/test/scala/docs/stream/operators/sink/HeadOption.scala:19]: Use `foreach` for sideffect only on Future

**Review by johanandren — COMMENTED** [review]
Looks good, two small improvement suggestions added.

File: akka-docs/src/test/scala/docs/stream/operators/sink/HeadOption.scala (hunk 2 above)
Comment (muskan3006) [reply akka-docs/src/test/scala/docs/stream/operators/sink/HeadOption.scala:19]: You are suggesting to use foreach in place of map?

Comment (johanandren) [reply akka-docs/src/test/scala/docs/stream/operators/sink/HeadOption.scala:19]: Yes. `map` transforms a `Future[T]` to a `Future[U]` given a `T => U` function, but in this case we just want to react/side effect on future completion so it is somewhat non-idiomatic to use `map`, that's what `Future.foreach` is meant for.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Review by johanandren — APPROVED** [review]
LGTM (I added a parenthesises pair after PR validation complained about that)

**Comment by muskan3006** [comment]
> LGTM (I added a parenthesises pair after PR validation complained about that)

Thanks

**Review by octonato — APPROVED** [review]

**Comment by octonato** [comment]
Thanks @muskan3006

**Pull request merged**